symphony-bdk-core/src/main/java/com/symphony/bdk/core/retry/RetryWithRecoveryBuilder.java

```diff
@@ -0,0 +1,176 @@
+package com.symphony.bdk.core.retry;
+
+import com.symphony.bdk.core.api.invoker.ApiException;
+import com.symphony.bdk.core.config.model.BdkRetryConfig;
+import com.symphony.bdk.core.retry.resilience4j.Resilience4jRetryWithRecovery;
+import com.symphony.bdk.core.util.function.ConsumerWithThrowable;
+import com.symphony.bdk.core.util.function.SupplierWithApiException;
+
+import org.apiguardian.api.API;
+
+import javax.ws.rs.ProcessingException;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Predicate;
+
+/**
+ * Builder class to facilitate the instantiation of a {@link RetryWithRecovery}.
+ *
+ * @param <T> the type to be returned by {@link RetryWithRecovery#execute()}.
+ */
+@API(status = API.Status.INTERNAL)
+public class RetryWithRecoveryBuilder<T> {
+  private String name;
+  private BdkRetryConfig retryConfig;
+  private SupplierWithApiException<T> supplier;
+  private Predicate<Throwable> retryOnExceptionPredicate;
+  private Predicate<ApiException> ignoreException;
+  private Map<Predicate<ApiException>, ConsumerWithThrowable> recoveryStrategies;
```

Comment:
What if we want to generalize this API to something else than our BDK usage? Having `ApiException` really stick the design to our usage

Reply:
Yes I agree, but this works because we `catch(ApiException e)`. If we have a param type E, we cannot `catch(E e)` anymore as it is forbidden by java. We could try to `catch(Throwable t)` and check `if (t instanceof E)` but this is not possible either. We could circumvent it but it makes things quite cumbersome : see https://stackoverflow.com/questions/1570073/java-instanceof-and-generics